apisix/core/request.lua

```diff
@@ -131,26 +135,37 @@ function _M.set_header(ctx, header_name, header_value)
         changed = a6_request.is_request_header_set()
     end
 
-    ngx.req.set_header(header_name, header_value)
+    if override then
+        ngx.req.set_header(header_name, header_value)
```

Comment:
For symmetry beauty, let's also localize this function too.

Reply:
Fixd.